File: core-mirai-ts/plugins/danmaku/api/space.js
// bilibili个人空间相关接口
const request = require('./request')

/**
 * 获取用户信息
 * @param mid 用户uid
 * @returns {Promise<void>}
 */
exports.spaceInfo = async function spaceInfo(mid) {
  return await request({
    url: 'http://api.bilibili.com/x/space/acc/info',
    data: {mid}
  })
}

/**
 * 获取用户信息(卡片)
 * @param mid 用户uid
 * @returns {Promise<void>}
 */
exports.spaceInfoWithFans = async function spaceInfoWithFans(mid) {
  return await request({
    url: 'http://api.bilibili.com/x/web-interface/card',
    data: {mid}
  })
}

File: core-nonebot/plugins/bilibili/api/Space.py
# !/usr/bin/env python3
# coding=utf-8

"""
Space Service Class
"""
from typing import Optional

from .BaseRequest import request
from ..exception import BiliUserNotFoundException


async def get_user_space_info(uid):
    """
    获取bilibili用户空间详细信息
    :param uid: 用户uid或者昵称
    :return: 用户资料卡json
    """

    response = await request('GET', F"http://api.bilibili.com/x/space/acc/info?mid={uid}")
    if not response.get('data'):
        raise BiliUserNotFoundException(uid)
    return response.get('data')


async def get_user_card_info(uid):
    url = F"http://api.bilibili.com/x/web-interface/card?mid={uid}"
    response = await request('GET', url)
    if not response.get('data'):
        raise BiliUserNotFoundException(uid)
    return response.get('data')